config/generic_update.py

```diff
@@ -0,0 +1,332 @@
+
+CHECKPOINTS_DIR = "/etc/sonic/checkpoints"
+CHECKPOINT_EXT = ".cp.json"
+class FileSystemRollbacker:
+    def __init__(self, patchorderer = PatchOrderer(), changeapplier = ChangeApplier(), configlocker = ConfigLocker(), configwrapper = ConfigWrapper()):
+        # NOTE: config-replacer receives a NoOpConfigLocker
+        self.configreplacer = PatchApplier(patchorder, changeapplier, NoOpConfigLocker(), configwrapper)
+        self.configlocker = configlocker
+
+    def rollback(self, checkpoint_name, verbose, dry_run):
+        if verbose:
+            # TODO: Implement verbose logging
+            raise NotImplementedError
+
+        if dry_run:
+            # TODO: Implement dry-run
+            raise NotImplementedError
+
+        __acquire_config_lock()
+        target_config = __get_checkpoint_content(checkpoint_name)
+
+        __config_replace(target_config)
+
+        __release_lock()
+
+    def checkpoint(self, checkpoint_name, verbose, dry_run):
+        if verbose:
+            # TODO: Implement verbose logging
+            raise NotImplementedError
+
+        if dry_run:
+            # TODO: Implement dry-run
+            raise NotImplementedError
+
+        __acquire_config_lock()
+        current_config = _get_current_config()
+
+        __save_checkpoint_content(current_config)
+
+        __release_lock()
+
+    def list_checkpoints(self, verbose, dry_run):
+        if verbose:
+            # TODO: Implement verbose logging
+            raise NotImplementedError
+
+        if dry_run:
+            # TODO: Implement dry-run
+            raise NotImplementedError
+
+        if not(__checkpoints_dir_exist()):
+            return []
+
+        return __get_checkpoint_names()
+
+    def delete_checkpoint(self, checkpoint_name, verbose, dry_run):
+        if verbose:
+            # TODO: Implement verbose logging
+            raise NotImplementedError
+
+        if dry_run:
+            # TODO: Implement dry-run
+            raise NotImplementedError
+
+        if not(__check_checkpoint_exists()):
+            raise Exception("Checkpoint does not exist")
+
+        __delete_checkpoint(checkpoint_name)
+
+    def __acquire_config_lock():
+        self.configlocker.acquire_lock()
+
+    def __release_config_lock():
+        self.configlocker.release_lock()
+
+    def __get_current_config():
+        self.configwrapper.get_current_config()
+
+    def __ensure_checkpoints_dir_exists(self):
+        os.makedirs(CHECKPOINTS_DIR, exist_ok=True)
+
+    def __save_checkpoint_content(name, content):
+        __ensure_checkpoints_dir_exists()
+        path = __get_checkpoint_full_path(name)
+        with open(path, "w") as fh:
+            fh.write(text)
+
+    def __get_checkpoint_content(name):
+        path = __get_checkpoint_full_path(name)
+        with open(path) as fh:
+            text = fh.read()
+            return json.loads(text)
+
+    def __get_checkpoint_full_path(name)
```

Comment:
>) [](start = 39, length = 1)

You miss `:` at the end of line.
It was detected by lgtm linter. Please treat every lgtm alert seriosly.